### ISSUE TITLE
Spring Config Server delivering config file but not activating profiles

### ISSUE THREAD

**Issue by CharliePrm88** [spring-cloud/spring-cloud-config#2045]
I'm migrating from 2.2 to 2.6 and I've seen that properties file are changed.

I have a situation that I don't think is covered (or maybe I misunderstood).

My situation is that I've a Config server that deliver:

```
application.properties --> containing properties shared by every microservices
application-dev.properties --> containing properties specific for dev profile
application-test.properties --> containing properties specific for test profile
application-production.properties --> containing properties specific for production profile
microservicesName.properties --> containing specific properties
microservicesName-mysql.properties --> if I use mysql profile
microservicesName-h2.properties --> if I use h2 profile
```
Before 2.4 I used to put in microservices application.properties path to fetch configuration from config server and in application.properties delivered by config server:

`spring.profiles.include=dev,mysql`

if for example I would like to activate for all microservices dev and mysql.

Now, I've converted include to groups. E.G.:

`spring.profiles.groups.myConf=dev,mysql`

(tried also with array notation --> spring.profiles.groups.myConf[0]=dev and [1]=mysql) and adding to every microservices

`spring.profiles.active=myConf`

Watching for the config server logs, it deliver the configuration properly. My problem is that I have some Bean annotated with profile E.g.:

`@Profile('h2')`
or
`@Profile('dev')`

These beans are not activated and application won't start (also, in startup-logs profiles=[myConf])

The only solution that I found is to insert in every microservice application.properties

`spring.profile.active=dev,mysql`
but this don't look very comfortable because I've a lot of microservices.

Which is the proper way?

P.s.: I tried also to not touch anything and use simply

`spring.config.use-legacy-processing=true`

in all .properties but Spring Boot application won't fetch configuration by Config Server.

**Comment by ryanjbaxter** [comment]
Can you provide a [complete, minimal, verifiable sample](https://stackoverflow.com/help/mcve) that reproduces the problem? It should be available as a GitHub (or similar) project or attached to this issue as a zip file.

**Comment by CharliePrm88** [comment]
Here an example. Please replace your absolute path in configServer's application.properties file.
Also, if you try to comment @Profile annotation in h2 service, everything works despite the h2 properties is included.

[Github.zip](https://github.com/spring-cloud/spring-cloud-config/files/7990585/Github.zip)

**Comment by CharliePrm88** [comment]
Actually, this bug looks It has been fixed with Spring Boot 2.6.6, Spring Framework 5.3.18 and Spring Cloud 2021.0.1
